workflows/cellranger-quant/run-cellranger.nf

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env nextflow
+nextflow.enable.dsl=2
+
+// basic parameters
+params.ref_dir = 's3://nextflow-ccdl-data/reference/homo_sapiens/ensembl-100'
+params.index_dir = 'cellranger_index'
+params.index_name = 'cdna'
+
+params.run_metafile = 's3://ccdl-scpca-data/sample_info/scpca-library-metadata.tsv'
+params.run_ids = "SCPCR000001" //comma separated list to be parsed into a list
+params.outdir = 's3://nextflow-ccdl-results/scpca/cellranger-quant'
+
+// build full paths
+params.index_path = "${params.ref_dir}/${params.index_dir}/${params.index_name}"
+
+process cellranger{
+  container '589864003899.dkr.ecr.us-east-1.amazonaws.com/scpca-cellranger:4.0.0'
+  publishDir "${params.outdir}", mode: 'copy'
+  label 'cpus_8'
+  label 'bigdisk'
+  input:
+    tuple val(id), val(samples), path(fastq_dir)
+    path index
+  output:
+    path output_id
+  script:
+    output_id = "${id}-${index}"
+    """
+    cellranger count \
+      --id=${output_id} \
+      --transcriptome=${index} \
+      --fastqs=${fastq_dir} \
+      --sample=${samples} \
+      --localcores=${task.cpus} \
+      --localmem=${task.memory.toGiga()}
+    """
+}
+
+def getCRsamples(filelist){
+  // takes a string with semicolon separated file names
+  // returns just the 'sample info' portion of the file names,
+  // as cellranger would interpret them, comma separated
+  fastq_files = filelist.tokenize(';')
+  samples = []
+  fastq_files.each{
+    // append sample names to list, using regex to extract element before S001, etc.
+    // [0] for the first match set, [1] for the first extracted element
+    samples << (it =~ /^(.+)_S.+_L.+_R.+.fastq.gz$/)[0][1]
+  }
+  // set to remove dup
+  return samples.toSet().join(',')
```

Comment:
Yeah, I will fix this comment to be more clear. The list of samples we made by pulling a section out of the file names will include duplicates, as there are multiple read files for each sample. I could have just used the Read1 files, but for some reason I didn't. 

Using the `.toSet()` method converts from a `list` to a `set`; a `set` can only have one of each item, so this is a quick way to deduplicate the list before joining it into a comma separated list the way cellranger is going to need it.

Reply:
Okay, that cleared this step up for me, thank you. 

Since you stated that, 
> I could have just used the Read1 files, but for some reason I didn't.

Do you think that using just the Read1 files would be slightly faster than the method you used above?

Reply:
Yes, very very slightly slower, most likely. But these lists are only going to be a couple dozen files long at the outside, so the difference will be impossible to measure.  I was a bit worried (still am) about what kind of variation there might be in file naming, and this should avoid most. I think all 10X files will have the format I coded into the regex, but I don't know if there is variation in things like `R1` vs `R01`.

Reply:
Gotcha 👍 , well I am sure we will find out!